ui/components/app/assets/token-list/token-list.tsx

```diff
@@ -33,17 +37,73 @@ export default function TokenList({
   const { tokenSortConfig } = useSelector(getPreferences);
   const selectedAccount = useSelector(getSelectedAccount);
   const conversionRate = useSelector(getConversionRate);
-  const nativeTokenWithBalance = useNativeTokenBalance();
   const shouldHideZeroBalanceTokens = useSelector(
     getShouldHideZeroBalanceTokens,
   );
   const contractExchangeRates = useSelector(
     getTokenExchangeRates,
     shallowEqual,
   );
-  const { tokensWithBalances, loading } = useAccountTotalFiatBalance(
+  const chains = useSelector(getChains);
+
+  if (!chains || chains.length === 0) {
+    return (
+      <Box
+        display={Display.Flex}
+        alignItems={AlignItems.center}
+        justifyContent={JustifyContent.center}
+        padding={7}
+        data-testid="token-list-loading-message"
+      >
+        {t('loadingTokens')}
+      </Box>
+    );
+  }
+
+  return (
+    <div>
+      <>{nativeToken}</>
+      {chains.map((chain) => (
+        <TokenListForChain
+          key={chain.network.chainId}
+          chain={chain}
+          onTokenClick={onTokenClick}
+          tokenSortConfig={tokenSortConfig}
+          selectedAccount={selectedAccount}
+          conversionRate={conversionRate}
+          contractExchangeRates={contractExchangeRates}
+          shouldHideZeroBalanceTokens={shouldHideZeroBalanceTokens}
+        />
+      ))}
```

Comment:
Rather than mapping over chains and rendering a separate list for each chain and pinning the `nativeToken` to the top, I would suggest doing some data manipulation first. I think it would make sense to move the sortedAssets `useMemo` function up to this level, before the return.

This will allow us to:

1. Make this into a flat array first, so that we don't have a separate `TokenList` for each chain. We should instead have a single `TokenList`, which is correctly sorted and filtered for all tokens for all chains

2. Allow us to include the native tokens for each chain in the sorted/filtered array, rather than pinning them as a fragment to the top as we are here.